common/network-common/src/main/java/org/apache/spark/network/client/TransportClient.java

```diff
@@ -141,26 +141,14 @@ public void fetchChunk(
     StreamChunkId streamChunkId = new StreamChunkId(streamId, chunkIndex);
     handler.addFetchRequest(streamChunkId, callback);
 
-    channel.writeAndFlush(new ChunkFetchRequest(streamChunkId)).addListener(future -> {
```

Comment:
Are the changes to these `.addListener()` calls primarily cleanup / refactoring? Is the intent to reduce the amount of _new_ duplicate code which would otherwise be added to `uploadStream` in this file?

Reply:
yes exactly.  Marcelo asked for this refactoring in his review -- there was already a ton of copy-paste, and instead of adding more made sense to refactor.  Shouldn't be any behavior change (there are minor changes that shouldn't matter ...  `channel.close()` happens before the more specific cleanup operations whereas it was in the middle previously, the `try` encompasses a bit more than before.)

Reply:
Thanks for explaining. I guess the re-ordering of `channel.close()` and the `handler` operations is safe because the handler doesn't hold references to the channel / otherwise does not interact with it (and doesn't hold references to objects tied to channel lifecycle (like buffers))?